File: backend/app/schemas/featured_media.py
from pydantic import BaseModel, ConfigDict, AnyUrl
from uuid import UUID
from datetime import datetime
from typing import Optional
from app.models.enums import Platform, MediaType

class FeaturedMediaBase(BaseModel):
    title: str
    caption: Optional[str] = None
    instagram_url: Optional[str] = None
    video_url: Optional[str] = None
    cloudinary_public_id: Optional[str] = None
    display_order: int
    is_active: bool = True

class FeaturedMediaCreate(FeaturedMediaBase):
    platform: Platform = Platform.instagram
    media_type: MediaType = MediaType.reel

class FeaturedMediaUpdate(FeaturedMediaBase):
    title: Optional[str] = None
    display_order: Optional[int] = None
    is_active: Optional[bool] = None

class FeaturedMediaResponse(FeaturedMediaBase):
    id: UUID
    platform: Platform
    media_type: MediaType
    created_at: datetime
    updated_at: datetime
    
    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/featuredMedia.ts
export type Platform = "INSTAGRAM" | "YOUTUBE" | "SELF_HOSTED";
export type MediaType = "REEL" | "SHORTS" | "TESTIMONIAL";

export interface FeaturedMedia {
  id: string;
  platform: Platform;
  media_type: MediaType;
  title: string;
  caption: string | null;
  instagram_url: string | null;
  video_url: string | null;
  cloudinary_public_id: string | null;
  display_order: number;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}
